sklearn/decomposition/dict_learning.py

```diff
@@ -282,7 +282,7 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
             algorithm,
             regularization=regularization, copy_cov=copy_cov,
             init=init[this_slice] if init is not None else None,
-            max_iter=max_iter)
+            max_iter=max_iter, n_jobs=n_jobs)
```

Comment:
> This is not going to work as it can trigger double Parallel pool.
Thanks a lot for pointing that out.

I also agree that we should collapse _sparse_encode and sparse_encode. Please have a look at my comment below.